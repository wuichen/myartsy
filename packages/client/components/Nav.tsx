import React from "react"
import styled, { createGlobalStyle } from "styled-components"
import colors from "../Assets/Colors"
import Icon from "./Icon"

const Nav = styled.div`
  border-bottom: 1px solid ${colors.grayRegular};
  display: flex;
  height: 70px;
  align-content: center;
`

const NavIcon = styled.a`
  display: inline-block;
  font-size: 32px;
  padding: 15px 0 15px 22px;
  margin-right: 10px;
  flex: 0;
`

/* stylelint-disable-next-line */
const GlobalStyle = createGlobalStyle<{ suppressMultiMountWarning: boolean }>`
  body {
    margin: 0;
  }
`

export interface NavBarProps extends React.Props<HTMLDivElement> {
  logoLink?: string
}

const NavBar: React.SFC<NavBarProps> = props => (
  <Nav>
    <GlobalStyle suppressMultiMountWarning />
    <NavIcon href={props.logoLink}>
      <Icon name="logotype" color="black" fontSize="40px" />
    </NavIcon>
    {props.children}
  </Nav>
)

export default NavBar
