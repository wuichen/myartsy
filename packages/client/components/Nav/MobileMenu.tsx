import { garamond } from "Assets/Fonts"
import React from "react"
import styled, { StyledFunction } from "styled-components"
import StyledTextLink from "../TextLink"

interface Props {
  navHeight?: string
}

export const MobileMenu: React.SFC<Props> = props => {
  return (
    <MobileMenuContainer {...props}>
      <StyledTextLink color="white" href="/">
        Home
      </StyledTextLink>
      <StyledTextLink color="white" href="/institutions">
        Museums
      </StyledTextLink>
      <StyledTextLink color="white" href="/artists">
        Artists
      </StyledTextLink>
      <StyledTextLink color="white" href="/articles">
        Articles
      </StyledTextLink>
      <StyledTextLink color="white" href="/shows">
        Show Guide
      </StyledTextLink>
      <StyledTextLink color="white" href="/art-fairs">
        Art Fairs
      </StyledTextLink>
      <StyledTextLink color="white" href="/galleries">
        Galleries
      </StyledTextLink>
      <StyledTextLink color="white" href="/auctions">
        Auctions
      </StyledTextLink>
    </MobileMenuContainer>
  )
}

MobileMenu.defaultProps = {
  navHeight: "0px",
}

const Div: StyledFunction<Props & React.HTMLProps<HTMLDivElement>> = styled.div
const MobileMenuContainer = Div`
  ${garamond("s19")};
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  width: 100%;
  background-color: black;
  left: 0;
  padding: 20px;
  margin-top: ${props => props.navHeight};
  ${StyledTextLink} {
    display: inline-block;
    width: 50%;
    line-height: 2;
  }
`
