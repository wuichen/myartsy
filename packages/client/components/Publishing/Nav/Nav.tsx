import { Flex, Serif } from "@artsy/palette"
import { PartnerInline } from "Components/Publishing/Partner/PartnerInline"
import React, { ReactNode } from "react"
import Waypoint from "react-waypoint"
import styled from "styled-components"
import { Media } from "Utils/Responsive"

interface Props extends React.HTMLProps<HTMLDivElement> {
  backgroundColor?: string
  canFix?: boolean
  children?: ReactNode
  color?: string
  sponsor?: any
  title?: string
  transparent?: boolean
  isSlideOpen?: boolean
}

interface State {
  isFixed: boolean
}

export class Nav extends React.Component<Props, State> {
  static defaultProps = {
    backgroundColor: "black",
    canFix: true,
    color: "white",
    transparent: false,
  }

  state = {
    isFixed: false,
  }

  setPosition = (isFixed: boolean) => {
    const { canFix } = this.props
    const currentPosition = this.state.isFixed

    if (canFix && isFixed !== currentPosition) {
      this.setState({ isFixed })
    }
  }

  render() {
    const {
      backgroundColor,
      children,
      color,
      sponsor,
      className,
      canFix,
      transparent,
      title,
      isSlideOpen,
    } = this.props
    const { isFixed } = this.state

    return (
      <div>
        <NavContainer
          backgroundColor={backgroundColor}
          className={className}
          color={color}
          isFixed={canFix && isFixed}
          transparent={!isFixed && transparent}
          isSlideOpen={isSlideOpen}
        >
          <PartnerInline
            url={sponsor && sponsor.partner_logo_link}
            logo={sponsor && sponsor.partner_condensed_logo}
            color={color}
            margin="0 10px"
          />

          <Media greaterThan="xs">
            <Title size="5" color={color} weight="semibold" textAlign="center">
              {title ? title : <a href="/articles">Artsy Editorial</a>}
            </Title>
          </Media>
          {children}
        </NavContainer>

        <Waypoint
          onEnter={() => this.setPosition(false)}
          onLeave={() => this.setPosition(true)}
        />
      </div>
    )
  }
}

export const NavContainer = styled(Flex)<{
  backgroundColor: string
  color: string
  transparent: boolean
  isFixed: boolean
  isSlideOpen: boolean
}>`
  background-color: ${props =>
    props.transparent ? "transparent" : props.backgroundColor};
  border-bottom: 1px solid ${props => props.color};
  position: relative;
  height: 52px;
  width: 100%;
  z-index: ${props => (props.isSlideOpen ? -1 : 10)};

  ${props =>
    props.transparent &&
    !props.isFixed &&
    `
    position: absolute;
    top: 0;
  `};

  ${props =>
    props.isFixed &&
    `
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  `};
`

const Title = styled(Serif)<{ color: string }>`
  position: absolute;
  width: 100%;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  align-items: center;
  display: flex;
  justify-content: center;
  color: ${props => props.color};

  a {
    color: ${props => props.color};
    text-decoration: none;
    z-index: 10;
  }
`
