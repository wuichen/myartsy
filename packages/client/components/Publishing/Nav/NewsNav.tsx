import React from "react"
import styled from "styled-components"
import colors from "../../../Assets/Colors"
import { pMedia } from "../../Helpers"
import { NewsDateHeader, NewsText } from "../News/NewsDateHeader"

interface Props {
  date?: string
  positionTop?: number
}

export const NewsNav: React.SFC<Props> = props => {
  const { date, positionTop } = props
  return (
    <NewsNavContainer positionTop={positionTop}>
      <MaxWidthContainer>
        {date && <NewsDateHeader date={date} />}
        <Title>News</Title>
      </MaxWidthContainer>
    </NewsNavContainer>
  )
}

const Title = styled(NewsText)`
  position: absolute;
  left: 30px;
  ${pMedia.sm`
    left: 20px;
  `};
`

const MaxWidthContainer = styled.div`
  position: relative;
  max-width: 780px;
  margin: auto;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 26px;
  ${pMedia.sm`
    min-height: 16px;
  `};
`

const NewsNavContainer = styled.div`
  position: fixed;
  top: ${(props: Props) => (props.positionTop ? props.positionTop : 0)}px;
  left: 0;
  right: 0;
  border-bottom: 1px solid ${colors.grayRegular};
  padding: 10px 0;
  background: white;
  z-index: 11;
`
