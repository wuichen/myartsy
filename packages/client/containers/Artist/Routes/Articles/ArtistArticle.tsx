import { Col, Row } from "@artsy/palette"
import React, { SFC } from "react"
import { Media } from "Utils/Responsive"

import {
  Box,
  Flex,
  Image,
  Sans,
  Separator,
  Serif,
  Spacer,
} from "@artsy/palette"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
  lastChild: boolean
}

export const ArticleItem: SFC<ArticleItemProps> = props => {
  return (
    <>
      <Media at="xs">
        <SmallArticleItem {...props} />
      </Media>
      <Media greaterThan="xs">
        <LargeArticleItem {...props} />
      </Media>
    </>
  )
}

const LargeArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title, lastChild } = props

  return (
    <>
      <Row
        style={{ cursor: "pointer" }}
        onClick={() => {
          window.location.href = href
        }}
      >
        <Col sm={2}>
          <Date size="3" weight="medium">
            {date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={href} className="noUnderline">
                  {title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <a href={href}>
            <Image
              style={{
                height: "100px",
                width: "auto",
                maxWidth: 140,
                maxHeight: 280,
                objectFit: "fill",
              }}
              src={imageUrl}
              mb={2}
            />
          </a>
        </Col>
      </Row>
      {/* FIXME: Weird block height issue... */}
      {!lastChild && <Separator />}
      <Spacer mt={3} />
    </>
  )
}

const SmallArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title, lastChild } = props

  return (
    <>
      <Flex
        justifyContent="space-between"
        style={{ cursor: "pointer" }}
        onClick={() => {
          window.location.href = href
        }}
      >
        <Box pr={2}>
          <Date size="1" weight="medium">
            {date}
          </Date>
          <Spacer mb={0.5} />
          <Title size="2">
            <a href={href} className="noUnderline">
              {title}
            </a>
          </Title>
          <Spacer mb={0.5} />
          <Credit size="1" color="black60">
            {author}
          </Credit>
        </Box>
        <Image width="70px" height="100%" src={imageUrl} />
      </Flex>
      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      {!lastChild && <Separator />}
      <Spacer mt={3} />
    </>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
