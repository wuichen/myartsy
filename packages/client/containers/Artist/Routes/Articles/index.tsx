import { Articles_artist } from "__generated__/Articles_artist.graphql"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { ArtistArticlesRefetchContainer as Articles } from "./ArtistArticles"

export interface ArticlesRouteProps {
  artist: Articles_artist
}

export const ArticlesRoute: React.SFC<ArticlesRouteProps> = props => {
  return <Articles artist={props.artist} />
}

export const ArticlesRouteFragmentContainer = createFragmentContainer(
  ArticlesRoute,
  {
    artist: graphql`
      fragment Articles_artist on Artist {
        ...ArtistArticles_artist
      }
    `,
  }
)
