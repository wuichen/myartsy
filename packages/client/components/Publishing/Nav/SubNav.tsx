import { avantgarde } from "Assets/Fonts"
import React from "react"
import styled from "styled-components"
import StyledButton from "../../Buttons/Default"
import InvertedButton from "../../Buttons/Inverted"
import { pMedia } from "../../Helpers"
import Icon from "../../Icon"
import { MobileMenu } from "../../Nav/MobileMenu"
import { IconHamburger } from "../Icon/IconHamburger"

interface State {
  mobileNavIsOpen?: boolean
}

export class SubNav extends React.Component<null, State> {
  constructor(props) {
    super(props)
    this.state = {
      mobileNavIsOpen: false,
    }
  }

  toggleMobileNav = () => {
    this.setState({
      mobileNavIsOpen: !this.state.mobileNavIsOpen,
    })
  }

  render() {
    return (
      <SubNavContainer>
        <Link className="mlh-login" href="/log_in">
          <Login>Login</Login>
        </Link>
        <Link className="mlh-signup" href="/sign_up">
          <Signup>Signup</Signup>
        </Link>
        <MobileNavToggle>
          {this.state.mobileNavIsOpen ? (
            <Icon name="close" color="white" onClick={this.toggleMobileNav} />
          ) : (
            <IconHamburger color="white" onClick={this.toggleMobileNav} />
          )}
        </MobileNavToggle>
        {this.state.mobileNavIsOpen && <MobileMenu navHeight="46px" />}
      </SubNavContainer>
    )
  }
}

const Login = styled(InvertedButton)`
  ${avantgarde("s11")};
  height: 30px;
  width: 80px;
  border-radius: 2px;
  margin: 8px;
`
const Signup = styled(StyledButton)`
  ${avantgarde("s11")};
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;
  margin: 8px;
`
const Link = styled.a`
  text-decoration: none;
`
const MobileNavToggle = styled.div`
  display: none;
  margin-right: 5px;
  max-width: 32px;
`
const SubNavContainer = styled.div`
  display: flex;
  height: 100%;
  ${pMedia.sm`
    ${Login}, ${Signup} {
      display: none;
    }
    ${MobileNavToggle} {
      display: flex;
      align-items: center;
    }
  `};
`
