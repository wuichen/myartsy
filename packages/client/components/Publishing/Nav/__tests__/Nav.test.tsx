import { SponsoredArticle } from "Components/Publishing/Fixtures/Articles"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { Nav } from "../Nav"

describe("Nav", () => {
  it("renders a Nav", () => {
    const nav = renderer.create(<Nav />)
    expect(nav).toMatchSnapshot()
  })
  it("renders a transparent nav", () => {
    const nav = renderer.create(<Nav transparent />)
    expect(nav).toMatchSnapshot()
  })
  it("renders a sponsored nav", () => {
    const nav = renderer.create(<Nav sponsor={SponsoredArticle.sponsor} />)
    expect(nav).toMatchSnapshot()
  })
  it("includes a partner logo in a sponsored nav", () => {
    const nav = mount(<Nav sponsor={SponsoredArticle.sponsor} />)
    expect(nav.html()).toMatch("cloudfront")
    expect(nav.html()).toMatch("Wort-Bildmarke_negative")
  })
  it("renders a custom title", () => {
    const nav = mount(<Nav title="Custom Title" />)
    expect(nav.text()).toMatch("Custom Title")
  })

  it("setPosition sets the state if props.canFix", () => {
    const nav = mount(<Nav />) as any
    nav.instance().setPosition(true)
    expect(nav.state().isFixed).toBe(true)
  })

  it("setPosition does not set state if without props.canFix", () => {
    const nav = mount(<Nav canFix={false} />) as any
    nav.instance().setPosition(true)
    expect(nav.state().isFixed).toBe(false)
  })
})
