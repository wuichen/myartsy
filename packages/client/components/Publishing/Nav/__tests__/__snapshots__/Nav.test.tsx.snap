// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders a Nav 1`] = `
.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-weight: 600;
  font-size: 22px;
  line-height: 32px;
  color: white;
  text-align: center;
}

.c3 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  position: relative;
  color: white;
}

.c1 {
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c1 img {
  max-height: 34px;
  vertical-align: middle;
  -ms-interpolation-mode: bicubic;
}

.c1 .c6 {
  margin: 0 20px;
}

.c1 .c2 {
  margin: 0;
  vertical-align: middle;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: black;
  border-bottom: 1px solid white;
  position: relative;
  height: 52px;
  width: 100%;
  z-index: 10;
}

.c4 {
  position: absolute;
  width: 100%;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: white;
}

.c4 a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 10;
}

@media (max-width:720px) {
  .c1 img {
    max-height: 24px;
  }

  .c1 .c2 {
    font-size: 24px;
  }

  .c1 .c6 {
    margin: 0 10px;
  }
}

<div>
  <div
    className="c0"
    color="white"
  >
    <div
      className="PartnerInline c1"
    >
      <a
        href="/"
      >
        <div
          className="c2 c3"
        >
          
        </div>
      </a>
      
    </div>
    <div
      className="rrm-container rrm-greaterThan-xs"
    >
      <div
        className="c4 c5"
        color="white"
        fontFamily={
          Object {
            "fontFamily": "'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif",
            "fontWeight": 600,
          }
        }
        fontSize="22px"
      >
        <a
          href="/articles"
        >
          Artsy Editorial
        </a>
      </div>
    </div>
  </div>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
</div>
`;

exports[`Nav renders a sponsored nav 1`] = `
.c7 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-weight: 600;
  font-size: 22px;
  line-height: 32px;
  color: white;
  text-align: center;
}

.c3 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  position: relative;
  color: white;
}

.c5 {
  width: 15px;
  height: 15px;
}

.c1 {
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c1 img {
  max-height: 34px;
  vertical-align: middle;
  -ms-interpolation-mode: bicubic;
}

.c1 .c4 {
  margin: 0 20px;
}

.c1 .c2 {
  margin: 0;
  vertical-align: middle;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: black;
  border-bottom: 1px solid white;
  position: relative;
  height: 52px;
  width: 100%;
  z-index: 10;
}

.c6 {
  position: absolute;
  width: 100%;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: white;
}

.c6 a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 10;
}

@media (max-width:900px) {
  .c5 {
    width: 13px;
    height: 13px;
  }
}

@media (max-width:720px) {
  .c1 img {
    max-height: 24px;
  }

  .c1 .c2 {
    font-size: 24px;
  }

  .c1 .c4 {
    margin: 0 10px;
  }
}

<div>
  <div
    className="c0"
    color="white"
  >
    <div
      className="PartnerInline c1"
    >
      <a
        href="/"
      >
        <div
          className="c2 c3"
        >
          
        </div>
      </a>
      <svg
        className="c4 c5"
        version="1.1"
        viewBox="0 0 15 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="2"
        >
          <g
            fill="white"
            transform="translate(-101.000000, -18.000000)"
          >
            <g
              transform="translate(101.000000, 18.000000)"
            >
              <rect
                height="1"
                width="15"
                x="0"
                y="7"
              />
              <rect
                height="1"
                transform="translate(7.500000, 7.500000) rotate(-90.000000) translate(-7.500000, -7.500000) "
                width="15"
                x="0"
                y="7"
              />
            </g>
          </g>
        </g>
      </svg>
      <a
        href="http://artsy.net"
        onClick={[Function]}
        target="_blank"
      >
        <img
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FGEu3iYW6CQhbVxsjpOYwZQ%252FGKL_Wort-Bildmarke_negative_rgb%2B2.png&width=240&quality=80"
        />
      </a>
    </div>
    <div
      className="rrm-container rrm-greaterThan-xs"
    >
      <div
        className="c6 c7"
        color="white"
        fontFamily={
          Object {
            "fontFamily": "'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif",
            "fontWeight": 600,
          }
        }
        fontSize="22px"
      >
        <a
          href="/articles"
        >
          Artsy Editorial
        </a>
      </div>
    </div>
  </div>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
</div>
`;

exports[`Nav renders a transparent nav 1`] = `
.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-weight: 600;
  font-size: 22px;
  line-height: 32px;
  color: white;
  text-align: center;
}

.c3 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  position: relative;
  color: white;
}

.c1 {
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c1 img {
  max-height: 34px;
  vertical-align: middle;
  -ms-interpolation-mode: bicubic;
}

.c1 .c6 {
  margin: 0 20px;
}

.c1 .c2 {
  margin: 0;
  vertical-align: middle;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
  border-bottom: 1px solid white;
  position: relative;
  height: 52px;
  width: 100%;
  z-index: 10;
  position: absolute;
  top: 0;
}

.c4 {
  position: absolute;
  width: 100%;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: white;
}

.c4 a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 10;
}

@media (max-width:720px) {
  .c1 img {
    max-height: 24px;
  }

  .c1 .c2 {
    font-size: 24px;
  }

  .c1 .c6 {
    margin: 0 10px;
  }
}

<div>
  <div
    className="c0"
    color="white"
  >
    <div
      className="PartnerInline c1"
    >
      <a
        href="/"
      >
        <div
          className="c2 c3"
        >
          
        </div>
      </a>
      
    </div>
    <div
      className="rrm-container rrm-greaterThan-xs"
    >
      <div
        className="c4 c5"
        color="white"
        fontFamily={
          Object {
            "fontFamily": "'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif",
            "fontWeight": 600,
          }
        }
        fontSize="22px"
      >
        <a
          href="/articles"
        >
          Artsy Editorial
        </a>
      </div>
    </div>
  </div>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
</div>
`;
